src/services/collections.js

```diff
@@ -0,0 +1,137 @@
+const axios = require("axios");
+
+const mockCollections = {
+  collections: [
+    {
+      collectionId: 123,
+      collectionName: "1st Gen Pokemon Cards",
+      collectionDescription: "My collection of 1st gen cards",
+      categoryId: 19,
+      categoryName: "Pokemon Cards",
+      userId: 12,
+      collectionCreationDate: "2021-09-23T01:22:47.541Z",
+      coverImages: [
+        {
+          url: "https://cdn.vox-cdn.com/thumbor/eFEHo8eygHajtwShwT9e_jf7c-c=/0x0:1920x1080/1200x800/filters:focal(722x227:1028x533)/cdn.vox-cdn.com/uploads/chorus_image/image/69323002/Screen_Shot_2021_05_21_at_9.54.00_AM.0.jpeg",
+        },
+        { url: "https://static.wikia.nocookie.net/pokemon/images/4/49/Ash_Pikachu.png/revision/latest?cb=20200405125039" },
+        { url: "https://cdn2.bulbagarden.net/upload/thumb/4/49/Ash_Pikachu.png/1200px-Ash_Pikachu.png" },
+      ],
+    },
+  ],
+};
+
+const mockCollection = {
+  collectionId: 123,
+  collectionName: "1st Gen Pokemon Cards",
+  collectionDescription: "My collection of 1st gen cards",
+  categoryId: 19,
+  categoryName: "Pokemon Cards",
+  userId: 12,
+  collectionCreationDate: "2021-09-23T01:22:47.541Z",
+};
+
+const mockPostCollectionResponse = {
+  collectionId: 123,
+};
+
+export const getCollections = async (categoryId, userId, offset, limit) => {
```

Comment:
Can consider using the following naming convention

If fetching multiple resources using a specific ID filter: `getCollectionsByCategoryId`
or to fetch a single resource: `getCollectionById`

If we want all without any Id filter, maybe can use `getAllCollections`

Reply:
I think we can use `getCollections` for all the above for now, because both `categoryId` and `userId` are optional. My thought process was we just pass whatever the user inputs (could be `null`), and the server handles the job of building the SQL query based on that, i.e. if `userId` is null, then collections will be filtered by `categoryId` only.